balancer/weightedtarget/weightedtarget_test.go

```diff
@@ -1077,38 +1098,45 @@ func (s) TestBalancerGroup_SubBalancerTurnsConnectingFromTransientFailure(t *tes
 
 	// Set both subconn to TransientFailure, this will put both sub-balancers in
 	// transient failure.
-	scConnErr := errors.New("subConn connection error")
+	wantSubConnErr := errors.New("subConn connection error")
 	wtb.UpdateSubConnState(sc1, balancer.SubConnState{
 		ConnectivityState: connectivity.TransientFailure,
-		ConnectionError:   scConnErr,
+		ConnectionError:   wantSubConnErr,
 	})
 	<-cc.NewPickerCh
 	wtb.UpdateSubConnState(sc2, balancer.SubConnState{
 		ConnectivityState: connectivity.TransientFailure,
-		ConnectionError:   scConnErr,
+		ConnectionError:   wantSubConnErr,
 	})
-	p := <-cc.NewPickerCh
 
-	for i := 0; i < 5; i++ {
-		if _, err := p.Pick(balancer.PickInfo{}); err == nil || !strings.Contains(err.Error(), scConnErr.Error()) {
-			t.Fatalf("want pick error %q, got error %q", scConnErr, err)
-		}
-	}
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
 
-	// Set one subconn to Connecting, it shouldn't change the overall state.
-	wtb.UpdateSubConnState(sc1, balancer.SubConnState{ConnectivityState: connectivity.Connecting})
-	select {
-	case <-time.After(100 * time.Millisecond):
-	case <-cc.NewPickerCh:
-		t.Fatal("received new picker from the LB policy when expecting none")
-	}
+	if err := cc.WaitForPicker(ctx, func(p balancer.Picker) error {
+		for i := 0; i < 5; i++ {
+			if _, err := p.Pick(balancer.PickInfo{}); !reflect.DeepEqual(err, fmt.Errorf("last connection error: subConn connection error")) {
+				return fmt.Errorf("want pick error %q, got error %q", wantSubConnErr, err)
+			}
+		}
+		// Set one subconn to Connecting, it shouldn't change the overall state.
```

Comment:
I want an extra pair of eyes here. The test is supposed to 1. update subConn state to TransientFailure and check for pick err. 2. change subconn state to Connecting ans ensure no new Picker can be created from this LB policy and also try picking for err

Reply:
This was not the intended TO-DO. what we wanted to do is to *not* use DeepEquals [(why?)](http://goto.google.com/go-style/decisions#types-of-equality)